### PR TITLE
feat: expandable pills and tool workflow styling from Ozwell widget

### PR DESCRIPTION
**Pull request by aditya-damerla128**
Closes #290.

Brings the Ozwell chat widget's pill patterns into the UI package as themeable components.

## What's added

- **PillSelect** (`src/components/PillSelect/`) — the expandable option selector (the reasoning-mode capsule that expands into a segmented control). Collapsed pill → click → options → pick → collapses back. Any number of options, fully themeable via design tokens. Closes on click-outside / Escape.

pillselect-collapsed
pillselect-expanded

- **ThinkingBlock** (in `AIMessage`) — reasoning pill showing "Thinking" with a pulsing dot while streaming, then "Thought for Xs" once done. Click to expand/collapse the reasoning text.

thinking-active
thinking-complete
thinking-expanded

- **MCPToolCall** — restyled as a status pill header + a capped-width detail box:
  - Friendly tool names with underscores stripped and tense shifted (`create_patient` → "Creating patient" → "Created patient").
  - Green (success) / red (error) color scheme kept the same as before.
  - Result summary and resource links always visible in the box; raw parameters toggle open/closed with a smooth animation.
- **CollapsiblePill** (`src/components/AI/CollapsiblePill.tsx`) — shared primitive behind the thinking + tool pills.

tool-error
tool-running
tool-success

## Checks
- Typecheck, lint, prettier: clean
- Tests: 15/15 pass
- Storybook stories added/updated for all new states

All components use mieweb/ui design tokens (light + dark mode).

### CODE REVIEW

File: src/components/AI/MCPToolCall.tsx [hunk 1]
```diff
@@ -703,103 +689,102 @@ export function MCPToolCallDisplay({
...
-          {collapsible && showParameters && toolCall.parameters.length > 0 && (
-            <button
-              onClick={() => setShowDetails(!showDetails)}
-              className="mt-2 flex items-center gap-1 text-xs text-neutral-600 hover:text-neutral-700 dark:text-neutral-400 dark:hover:text-neutral-300"
+        {/* Show Details Toggle */}
+        {collapsible && showParameters && toolCall.parameters.length > 0 && (
+          <button
+            onClick={() => setShowDetails(!showDetails)}
+            className="flex items-center gap-1 text-xs text-neutral-600 hover:text-neutral-700 dark:text-neutral-400 dark:hover:text-neutral-300"
+          >
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/MCPToolCall.tsx]: Fixed — this button was removed entirely. The tool call is now pill-first: clicking the status pill reveals the whole box (result + raw params) in one action, so there's no separate untyped `<button>` anymore.

File: src/components/AI/CollapsiblePill.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  children,
+}: CollapsiblePillProps) {
+  const [internalOpen, setInternalOpen] = React.useState(defaultOpen);
+  const isOpen = controlledOpen !== undefined ? controlledOpen : internalOpen;
+
+  const toggle = () => {
+    const next = !isOpen;
+    setInternalOpen(next);
+    onOpenChange?.(next);
+  };
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/CollapsiblePill.tsx:37]: Fixed — CollapsiblePill now only writes internal state when uncontrolled (added an `isControlled` guard). When a controlled `open` prop is passed, the consumer fully drives the state.

File: src/components/PillSelect/PillSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  function handleSelect(option: PillSelectOption) {
+    if (option.disabled) return;
+    if (controlledValue === undefined) setInternalValue(option.value);
+    onValueChange?.(option.value);
+    setExpanded(false);
+  }
+
+  const collapsedLabel = [label, selectedOption?.label ?? value]
+    .filter(Boolean)
+    .join(': ');
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/PillSelect/PillSelect.tsx]: Fixed — the collapsed label is now built explicitly: `label && valuePart ? `${label}: ${valuePart}` : (label || valuePart)`, so there's no trailing colon when the selected value is empty.

File: src/components/PillSelect/PillSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      ) : (
+        <button
+          type="button"
+          disabled={disabled}
+          onClick={() => setExpanded(true)}
+          className={cn(
+            'border-border bg-muted text-foreground inline-flex items-center rounded-full border px-3 py-0.5 text-sm transition-colors',
+            'hover:bg-muted/80 focus-visible:ring-ring focus:outline-none focus-visible:ring-2',
+            disabled && 'cursor-not-allowed opacity-50'
+          )}
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/PillSelect/PillSelect.tsx]: Fixed — the collapsed pill is now disabled when there are no options (`disabled={disabled || !hasOptions}`).

File: src/components/AI/CollapsiblePill.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      <button
+        type="button"
+        onClick={toggle}
+        aria-expanded={isOpen}
+        className={cn(
+          'inline-flex items-center gap-1.5 border font-medium transition-colors focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-1',
+          density === 'condensed'
+            ? 'rounded-[8px] px-2 py-0.5 text-[10px]'
+            : 'rounded-[10px] px-[11px] py-1 text-[11px]',
+          pillClassName
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/CollapsiblePill.tsx]: Fixed — added `rounded-[8px]`, `rounded-[10px]`, and `px-[11px]` to `miewebUISafelist` in src/tailwind-preset.ts.

File: src/components/AI/MCPToolCall.tsx [hunk 6]
```diff
@@ -703,103 +689,102 @@ export function MCPToolCallDisplay({
...
 
-          {/* Technical Details (collapsed by default) */}
-          {showDetails && showParameters && toolCall.parameters.length > 0 && (
+        {/* Technical Details — smooth expand/collapse */}
+        {showParameters && toolCall.parameters.length > 0 && (
+          <div
+            className={cn(
+              'overflow-hidden transition-[max-height,opacity] duration-300 ease-in-out',
+              showDetails ? 'max-h-[500px] opacity-100' : 'max-h-0 opacity-0'
+            )}
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/MCPToolCall.tsx]: Fixed — added `max-h-0`, `max-h-[500px]`, and `transition-[max-height,opacity]` to `miewebUISafelist`.

File: src/components/AI/AIMessage.tsx [hunk 7]
```diff
@@ -161,6 +162,61 @@ function AITypingIndicator({ className }: { className?: string }) {
...
+    <div data-slot="ai-message-thinking">
+      <style>{`@keyframes thinking-pulse { 0%, 100% { opacity: 0.3; } 50% { opacity: 1; } }`}</style>
+      <CollapsiblePill
+        label={label}
+        leadingIcon={dot}
+        defaultOpen={!defaultCollapsed}
+        pillClassName="bg-violet-50 border-violet-200 text-violet-600 hover:bg-violet-100 dark:bg-violet-950/30 dark:border-violet-800 dark:text-violet-400 dark:hover:bg-violet-950/50 focus-visible:ring-violet-500"
+      >
+        <div className="border-l-2 border-violet-200 pl-3 text-[13px] leading-relaxed text-neutral-600 italic dark:border-violet-700 dark:text-neutral-400">
+          {text}
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/AIMessage.tsx:211]: Fixed — added the violet thinking-pill classes to `miewebUISafelist` (plus the green/red status-pill colors, which were also missing from the safelist).

File: src/components/AI/MCPToolCall.tsx [hunk 8]
```diff
@@ -664,17 +637,33 @@ function getParameterSummary(
...
-  /** Whether to show parameters (in detailed view) */
+  /** Compact (condensed) sizing — tighter pill and box */
+  compact?: boolean;
+  /** Whether to show the raw parameters ("Show details") inside the box */
   showParameters?: boolean;
-  /** Whether the component is collapsible to show details */
+  /** Whether the raw parameters can be collapsed/expanded inside the box */
   collapsible?: boolean;
-  /** Default collapsed state (true = hide details) */
+  /** Default collapsed state (true = pill only, box hidden until clicked) */
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/MCPToolCall.tsx]: Fixed in c0c458b. The MCPToolCallDisplay prop docs now describe the actual behavior: showParameters controls whether raw parameters are included in the detail box, and collapsible controls whether the pill expands/collapses the whole detail box.

File: src/components/AI/CollapsiblePill.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+          }
+        />
+      </button>
+      {children && (
+        <div
+          className={cn(
+            'overflow-hidden transition-[max-height,opacity] duration-300 ease-in-out',
+            isOpen ? 'max-h-[500px] opacity-100' : 'max-h-0 opacity-0'
+          )}
+        >
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/CollapsiblePill.tsx:80]: Fixed in c0c458b. The collapsible content now gets aria-hidden when closed and inert={!isOpen || undefined}, matching the ServiceAccordion pattern, with aria-controls wiring the pill button to the panel.

File: src/components/AI/MCPToolCall.tsx [hunk 10]
```diff
@@ -691,140 +680,131 @@ export function MCPToolCallDisplay({
   collapsible = true,
   defaultCollapsed = true,
   compact,
+  hidden = false,
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/MCPToolCall.tsx:642]: Fixed in c0c458b. defaultCollapsed is now derived when omitted: success/error tool calls start expanded, while in-progress states stay collapsed. Explicit defaultCollapsed still overrides the status-based default.

File: src/components/AI/MCPToolCall.stories.tsx [hunk 11]
```diff
@@ -52,20 +52,27 @@ const meta: Meta<typeof MCPToolCallDisplay> = {
     },
     collapsible: {
       control: 'boolean',
-      description: 'Whether the technical details can be collapsed/expanded.',
+      description:
+        'Whether the raw parameters can be collapsed/expanded (the "Show details" toggle inside the box).',
       table: { defaultValue: { summary: 'true' } },
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/MCPToolCall.stories.tsx:44]: Fixed in c0c458b. The Storybook control text now says collapsible toggles the entire detail box, and the defaultCollapsed summary reflects the status-based default. I also removed redundant story args covered by that default.

File: src/components/PillSelect/PillSelect.tsx [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+    label && valuePart ? `${label}: ${valuePart}` : label || valuePart;
+
+  return (
+    <div ref={ref} className={cn('inline-flex', className)}>
+      {expanded ? (
+        <div
+          role="group"
+          aria-label={label}
+          className="border-border bg-muted inline-flex items-center rounded-full border p-0.5"
+        >
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/PillSelect/PillSelect.tsx]: Fixed in 1f1cb73. The expanded PillSelect group now uses a fallback accessible name (`Options`) when no label prop is provided.

File: src/components/AI/CollapsiblePill.tsx [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+        aria-expanded={isOpen}
+        aria-controls={children ? contentId : undefined}
+        title={title}
+        className={cn(
+          'inline-flex items-center gap-1.5 border font-medium transition-colors focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-1',
+          density === 'condensed'
+            ? 'rounded-[8px] px-2 py-0.5 text-[10px]'
+            : 'rounded-[10px] px-[11px] py-1 text-[11px]',
+          pillClassName
+        )}
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/CollapsiblePill.tsx:54]: Fixed in 1f1cb73. CollapsiblePill now includes `focus-visible:ring-ring` while keeping the existing ring width and offset; the ring utilities are also included in the safelist.

File: src/components/PillSelect/PillSelect.tsx [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+                aria-pressed={selected}
+                autoFocus={selected}
+                className={cn(
+                  'focus-visible:ring-ring rounded-full px-3 py-0.5 text-sm transition-colors focus:outline-none focus-visible:ring-2',
+                  selected
+                    ? 'bg-background text-foreground font-semibold shadow-sm'
+                    : 'text-muted-foreground hover:text-foreground font-normal',
+                  (disabled || option.disabled) && 'cursor-not-allowed opacity-40'
+                )}
+              >
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/PillSelect/PillSelect.tsx]: Fixed in 1f1cb73. The autoFocus prop was removed entirely, which avoids focusing disabled selected options and satisfies the repo lint rule against autofocus.

File: src/components/AI/MCPToolCall.tsx [hunk 15]
```diff
@@ -664,17 +637,33 @@ function getParameterSummary(
...
+    'bg-neutral-100 border-neutral-200 text-neutral-600 dark:bg-neutral-800 dark:border-neutral-700 dark:text-neutral-400',
+  running:
+    'bg-primary-50 border-primary-200 text-primary-800 dark:bg-primary-900/20 dark:border-primary-800 dark:text-primary-300',
+  success:
+    'bg-green-50 border-green-200 text-green-700 dark:bg-green-900/20 dark:border-green-800 dark:text-green-300',
+  error:
+    'bg-red-50 border-red-200 text-red-700 dark:bg-red-900/20 dark:border-red-800 dark:text-red-300',
+  cancelled:
+    'bg-neutral-100 border-neutral-200 text-neutral-400 opacity-60 dark:bg-neutral-800 dark:border-neutral-700',
+};
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/MCPToolCall.tsx:611]: Fixed in 4ec868f. The cancelled status pill now includes `dark:text-neutral-400`, matching the explicit dark-mode text class pattern used by the other status pills. `dark:text-neutral-400` was already present in the Tailwind safelist.

File: src/tailwind-preset.ts [hunk 16]
```diff
@@ -406,6 +406,56 @@ export const miewebUISafelist = [
...
+  'rounded-[8px]',
+  'rounded-[10px]',
+  'px-[11px]',
+  'max-h-0',
+  'max-h-[500px]',
+  'transition-[max-height,opacity]',
+  'overflow-y-auto',
+  'focus-visible:ring-ring',
+  'focus-visible:ring-offset-1',
+  // ThinkingBlock (violet reasoning pill)
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/tailwind-preset.ts:490]: Fixed in d7f5718. `text-[13px]` is now included in `miewebUISafelist`, so Tailwind CSS 3 consumers that do not scan package source will still retain the ThinkingBlock text size.

File: src/components/AI/AIMessage.tsx [hunk 17]
```diff
@@ -161,6 +162,61 @@ function AITypingIndicator({ className }: { className?: string }) {
...
+    : elapsed !== null && elapsed > 0
+      ? `Thought for ${elapsed}s`
+      : 'Thought';
+
+  const dot = streaming ? (
+    <span
+      aria-hidden="true"
+      className="inline-block h-1.5 w-1.5 rounded-full bg-violet-500 dark:bg-violet-400"
+      style={{ animation: 'thinking-pulse 1.2s ease-in-out infinite' }}
+    />
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/AIMessage.tsx:199]: Fixed in d7f5718. The thinking dot now uses Tailwind `animate-pulse` directly instead of the custom `thinking-pulse` inline animation.

File: src/components/AI/AIMessage.tsx [hunk 18]
```diff
@@ -161,6 +162,61 @@ function AITypingIndicator({ className }: { className?: string }) {
...
+  ) : null;
+
+  return (
+    <div data-slot="ai-message-thinking">
+      <style>{`@keyframes thinking-pulse { 0%, 100% { opacity: 0.3; } 50% { opacity: 1; } }`}</style>
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/AIMessage.tsx]: Fixed in d7f5718. The per-message `@keyframes thinking-pulse` style tag was removed along with the custom animation.

File: src/components/AI/CollapsiblePill.tsx [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+  className,
+  title,
+  children,
+}: CollapsiblePillProps) {
+  const [isOpen, setIsOpen] = React.useState(defaultOpen);
+  const contentId = React.useId();
+
+  const toggle = () => {
+    setIsOpen((open) => !open);
+  };
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/CollapsiblePill.tsx:37]: Fixed in 00cb74a. `CollapsiblePill` now syncs its open state when `defaultOpen` changes, so status-derived transitions such as `running` to `success` can auto-expand the details. Added `src/components/AI/CollapsiblePill.test.tsx` to cover the rerender path.

File: src/components/AI/CollapsiblePill.tsx [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+        <span>{label}</span>
+        <ChevronIcon
+          direction={isOpen ? 'down' : 'right'}
+          className={
+            density === 'condensed'
+              ? 'h-2.5 w-2.5 opacity-60'
+              : 'h-3 w-3 opacity-60'
+          }
+        />
+      </button>
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/CollapsiblePill.tsx:68]: Fixed in f2e4237. `CollapsiblePill` now only exposes disclosure semantics when it has children: no-content pills are disabled, omit `aria-expanded`/`aria-controls`, and do not render the chevron. Added coverage in `src/components/AI/CollapsiblePill.test.tsx` for the static no-content case.

File: src/components/PillSelect/PillSelect.tsx [hunk 21]
```diff
@@ -0,0 +1,110 @@
...
+    if (controlledValue === undefined) setInternalValue(option.value);
+    onValueChange?.(option.value);
+    setExpanded(false);
+  }
+
+  const hasOptions = options.length > 0;
+  const valuePart = selectedOption?.label ?? value;
+  const collapsedLabel =
+    label && valuePart ? `${label}: ${valuePart}` : label || valuePart;
+
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/PillSelect/PillSelect.tsx:143]: Fixed in c19f84d. `PillSelect` now falls back to `No options` when both `label` and selected value are empty, so the disabled empty-options control still has visible text and an accessible name. Added test coverage for this edge case in `src/components/PillSelect/PillSelect.test.tsx`.

File: src/components/PillSelect/PillSelect.tsx [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+  useClickOutside(ref, () => setExpanded(false), expanded);
+  useEscapeKey(() => setExpanded(false), expanded);
+
+  function handleSelect(option: PillSelectOption) {
+    if (disabled || option.disabled) return;
+    if (controlledValue === undefined) setInternalValue(option.value);
+    onValueChange?.(option.value);
+    setExpanded(false);
+  }
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/PillSelect/PillSelect.tsx:134]: Fixed in c19f84d. Added `src/components/PillSelect/PillSelect.test.tsx` covering the core interaction paths: empty fallback label, expand/select, disabled option behavior, controlled value display/callback, and Escape-to-close.

File: src/components/PillSelect/PillSelect.tsx [hunk 23]
```diff
@@ -0,0 +1,112 @@
...
+  );
+  const [expanded, setExpanded] = React.useState(false);
+  const ref = React.useRef<HTMLDivElement>(null);
+
+  const value = controlledValue !== undefined ? controlledValue : internalValue;
+  const selectedOption = options.find((o) => o.value === value);
+
+  useClickOutside(ref, () => setExpanded(false), expanded);
+  useEscapeKey(() => setExpanded(false), expanded);
+
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/PillSelect/PillSelect.tsx:128]: Fixed in a4c4ef1. PillSelect now uses a memoized close callback for useClickOutside and useEscapeKey, so the global listeners do not churn while the menu is expanded.

File: src/components/AI/MCPToolCall.tsx [hunk 24]
```diff
@@ -689,142 +678,136 @@ export function MCPToolCallDisplay({
...
-          {getToolIcon(toolCall.toolName)}
+      {/* Result Summary & Links */}
+      {toolCall.result && (
+        <div data-slot="ai-tool-result">
+          <ToolResultDisplay
+            result={toolCall.result}
+            onLinkClick={onLinkClick}
+          />
         </div>
+      )}
```
[the remark this answers is not in the document]
Comment (aditya-damerla128) [reply src/components/AI/MCPToolCall.tsx:692]: Fixed in a4c4ef1. Removed the unreachable showRawData prop and raw JSON branch from ToolResultDisplay instead of re-exposing that UI path.

**Review by garrity-miepub — DISMISSED** [review]
This update looks fantastic! Thank you for reworking it! I think this will make the component reusable in many different contexts.
